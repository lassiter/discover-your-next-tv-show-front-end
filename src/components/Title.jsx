import React, { Component } from 'react'
import styled from 'styled-components'

const Header = styled.h1`
  color: white;
  display: inline-flex;
`
const SubHeader = styled.h3`
  color: white;
  display: inline-flex;
`
const Span = styled.span`

`
const StartDate = styled.h3`
  color: darkgray;
  display: inline-flex;
  margin-left: 15px;
`

export default class Title extends Component {
  render() {
    if (this.props.name === this.props.originalName) {
      return (
        <>
          <Span><Header>{this.props.originalName}</Header><StartDate>({this.props.firstAirDate.slice(0,4)})</StartDate></Span>
        </>
      )
    } else {
      return (
        <>
          <Span><Header>{this.props.originalName}</Header><StartDate>({this.props.firstAirDate.slice(0,4)})</StartDate></Span>
          <SubHeader>{this.props.name}</SubHeader>
        </>
      )
    }
  }
}
