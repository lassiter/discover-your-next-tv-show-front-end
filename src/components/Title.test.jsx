import React from 'react';
import Title from './Title';
import renderer from 'react-test-renderer';

it('renders correctly with unmatching original name and name', () => {
  const name = "Ghost in the Shell"
  const originalName = "攻殻機動隊"
  const firstAirDate = "2002_10_1"
  const tree = renderer
    .create(<Title name={name} originalName={originalName} firstAirDate={firstAirDate}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with matching original name and name', () => {
  const name = "Ghost in the Shell"
  const originalName = "攻殻機動隊"
  const firstAirDate = "2002_10_1"
  const tree = renderer
    .create(<Title name={name} originalName={originalName} firstAirDate={firstAirDate}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});