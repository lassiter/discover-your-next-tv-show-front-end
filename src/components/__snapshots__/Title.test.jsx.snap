// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with matching original name and name 1`] = `
Array [
  <span
    className="sc-htpNat gPdQJU"
  >
    <h1
      className="sc-bdVaJa hiOOZY"
    >
      攻殻機動隊
    </h1>
    <h3
      className="sc-bxivhb dYyLZ"
    >
      (
      2002
      )
    </h3>
  </span>,
  <h3
    className="sc-bwzfXH iXQaeo"
  >
    Ghost in the Shell
  </h3>,
]
`;

exports[`renders correctly with unmatching original name and name 1`] = `
Array [
  <span
    className="sc-htpNat gPdQJU"
  >
    <h1
      className="sc-bdVaJa hiOOZY"
    >
      攻殻機動隊
    </h1>
    <h3
      className="sc-bxivhb dYyLZ"
    >
      (
      2002
      )
    </h3>
  </span>,
  <h3
    className="sc-bwzfXH iXQaeo"
  >
    Ghost in the Shell
  </h3>,
]
`;
